### ISSUE TITLE
CrmFetchKit.GetById fails for the Appointment entity.

### ISSUE THREAD

**Issue by jkovacs-dev** [thuld/CrmFetchKit#8]
When attempting to retrieve an Appointment entity from CRM using GetById (or GetByIdSync), the request fails with the message: `"'Appointment' entity doesn't contain attribute with Name = 'appointmentid'."`

Digging into the code, it would appear that the issue is caused by the following line in the function `buildGetByIdFetchXml` of soapXmlMessages.js

```javascript
 var idAttribute = entityname + 'id';
```

Unfortunately, the id field on the appointment entity is actually `activityid`.  I suspect this will also fail for other activity entity types as well.

**Comment by thuld** [comment]
I assume in this situation you neede to construct the fetchxml yourself and execute the query via FetchSync.

Daniel

Von meinem Windows Phone gesendet
________________________________
Von: Jonathan Kovacs<mailto:notifications@github.com>
Gesendet: ‎30.‎10.‎2015 17:58
An: thuld/CrmFetchKit<mailto:CrmFetchKit@noreply.github.com>
Betreff: [CrmFetchKit] CrmFetchKit.GetById fails for the Appointment entity. (#8)

When attempting to retrieve an Appointment entity from CRM using GetById (or GetByIdSync), the request fails with the message: `"'Appointment' entity doesn't contain attribute with Name = 'appointmentid'."`

Digging into the code, it would appear that the issue is caused by the following line in the function `buildGetByIdFetchXml` of soapXmlMessages.js

```javascript
 var idAttribute = entityname + 'id';
```

Unfortunately, the id field on the appointment entity is actually `activityid`.  I suspect this will also fail for other activity entity types as well.


---
Reply to this email directly or view it on GitHub:
https://github.com/thuld/CrmFetchKit/issues/8

**Comment by jkovacs-dev** [comment]
Yes, that is the workaround I'm using. To be honest, I can't really think
of a good way to 'fix' this so GetById would just work for both Activity
and non-activity entities. (Theoretically, you could query the CRM metadata
to get the id field, but that would kill performance.)  It might just be
worth updating the documentation for the method to note that it doesn't
work for activities.

Alternatively, it shouldn't be much work to add a
GetActivityById/GetActivityByIdSync method that did the right thing for
activity entities.  If you think that would be worthwhile, I can work it up
and send you a pull request.

**Comment by thuld** [comment]
Hello Jonathan,

It might ne an option to talke a look at the SDK. As fare as i remember contains the sdk an sample for JavaScript CURD Operation. In this code you might find a implementation for the SOAP retrieve operation. The retrieve operation works without the specification of the primary atteibute.

I am right now at a party and not able to look at it myself. In case you find the code for the retrieve operation an pull request is more than welcome.

In general you are right, i should update the documentation.

General question: Are you interested in becomimg an part of the developer team of the CrmFetchKit?

Daniel

Von meinem Windows Phone gesendet
________________________________
Von: Jonathan Kovacs<mailto:notifications@github.com>
Gesendet: ‎30.‎10.‎2015 20:44
An: thuld/CrmFetchKit<mailto:CrmFetchKit@noreply.github.com>
Cc: Daniel Thul<mailto:thuld@outlook.com>
Betreff: Re: [CrmFetchKit] CrmFetchKit.GetById fails for the Appointment entity. (#8)

Yes, that is the workaround I'm using. To be honest, I can't really think
of a good way to 'fix' this so GetById would just work for both Activity
and non-activity entities. (Theoretically, you could query the CRM metadata
to get the id field, but that would kill performance.)  It might just be
worth updating the documentation for the method to note that it doesn't
work for activities.

Alternatively, it shouldn't be much work to add a
GetActivityById/GetActivityByIdSync method that did the right thing for
activity entities.  If you think that would be worthwhile, I can work it up
and send you a pull request.


On 30 October 2015 at 13:26, Daniel Thul <notifications@github.com> wrote:

> I assume in this situation you neede to construct the fetchxml yourself
> and execute the query via FetchSync.
>
> Daniel
>
> Von meinem Windows Phone gesendet
> ________________________________
> Von: Jonathan Kovacs<mailto:notifications@github.com>
> Gesendet: ‎30.‎10.‎2015 17:58
> An: thuld/CrmFetchKit<mailto:CrmFetchKit@noreply.github.com>
> Betreff: [CrmFetchKit] CrmFetchKit.GetById fails for the Appointment
> entity. (#8)
>
> When attempting to retrieve an Appointment entity from CRM using GetById
> (or GetByIdSync), the request fails with the message: `"'Appointment'
> entity doesn't contain attribute with Name = 'appointmentid'."`
>
> Digging into the code, it would appear that the issue is caused by the
> following line in the function `buildGetByIdFetchXml` of soapXmlMessages.js
>
> ```javascript
> var idAttribute = entityname + 'id';
> ```
>
> Unfortunately, the id field on the appointment entity is actually
> `activityid`. I suspect this will also fail for other activity entity types
> as well.
>
>
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/thuld/CrmFetchKit/issues/8
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/thuld/CrmFetchKit/issues/8#issuecomment-152593995>.
>


---

**Comment by thuld** [comment]
Hello Jonathan, 

I just released version 3.3.2. This version uses
 the SOAP Retrieve method an this should solve our probelm with the 
activity issue (appointmentid vs. actvityid). 

Regards, 

Daniel

To: reply@reply.github.com; CrmFetchKit@noreply.github.com
From: thuld@outlook.com
Subject: AW: [CrmFetchKit] CrmFetchKit.GetById fails for the Appointment entity. (#8)
Date: Fri, 30 Oct 2015 21:30:39 +0100







Hello Jonathan,



It might ne an option to talke a look at the SDK. As fare as i remember contains the sdk an sample for JavaScript CURD Operation. In this code you might find a implementation for the SOAP retrieve operation. The retrieve operation works without the specification
 of the primary atteibute.



I am right now at a party and not able to look at it myself. In case you find the code for the retrieve operation an pull request is more than welcome.



In general you are right, i should update the documentation. 



General question: Are you interested in becomimg an part of the developer team of the CrmFetchKit?



Daniel



Von meinem Windows Phone gesendet



Von:
Jonathan Kovacs

Gesendet:
‎30.‎10.‎2015 20:44

An:
thuld/CrmFetchKit

Cc:
Daniel Thul

Betreff:
Re: [CrmFetchKit] CrmFetchKit.GetById fails for the Appointment entity. (#8)




Yes, that is the workaround I'm using. To be honest, I can't really think

of a good way to 'fix' this so GetById would just work for both Activity

and non-activity entities. (Theoretically, you could query the CRM metadata

to get the id field, but that would kill performance.) It might just be

worth updating the documentation for the method to note that it doesn't

work for activities.



Alternatively, it shouldn't be much work to add a

GetActivityById/GetActivityByIdSync method that did the right thing for

activity entities. If you think that would be worthwhile, I can work it up

and send you a pull request.





On 30 October 2015 at 13:26, Daniel Thul <notifications@github.com> wrote:



> I assume in this situation you neede to construct the fetchxml yourself

> and execute the query via FetchSync.

>

> Daniel

>

> Von meinem Windows Phone gesendet

> ________________________________

> Von: Jonathan Kovacs<mailto:notifications@github.com>

> Gesendet: ‎30.‎10.‎2015 17:58

> An: thuld/CrmFetchKit<mailto:CrmFetchKit@noreply.github.com>

> Betreff: [CrmFetchKit] CrmFetchKit.GetById fails for the Appointment

> entity. (#8)

>

> When attempting to retrieve an Appointment entity from CRM using GetById

> (or GetByIdSync), the request fails with the message: `"'Appointment'

> entity doesn't contain attribute with Name = 'appointmentid'."`

>

> Digging into the code, it would appear that the issue is caused by the

> following line in the function `buildGetByIdFetchXml` of soapXmlMessages.js

>

> ```javascript

> var idAttribute = entityname + 'id';

> ```

>

> Unfortunately, the id field on the appointment entity is actually

> `activityid`. I suspect this will also fail for other activity entity types

> as well.

>

>

> ---

> Reply to this email directly or view it on GitHub:

> https://github.com/thuld/CrmFetchKit/issues/8

>

> —

> Reply to this email directly or view it on GitHub

> <https://github.com/thuld/CrmFetchKit/issues/8#issuecomment-152593995>.

>

—

Reply to this email directly or 
view it on GitHub.
